api/pkg/apis/v1alpha1/model/skill.go

```diff
@@ -50,7 +50,8 @@ func (c SkillSpec) DeepEquals(other IDeepEquals) (bool, error) {
 	if !SlicesEqual(c.Bindings, otherSkillSpec.Bindings) {
 		return false, nil
 	}
-	if !SlicesEqual(c.Edges, otherSkillSpec.Edges) {
+
+	if !SlicesEqual(ExtractReferenceSlice(c.Edges), ExtractReferenceSlice(otherSkillSpec.Edges)) {
```

Comment:
Would it be better if we change the edge deepequal to expect an edge instead a pointer of edge?

Reply:
I agree. I have changed DeepEquals in SkillSpec, ModelSpec, EdgeSpec, ConnectionSpec to expect struct instead of pointer.